src/System.Private.Xml/src/System/Xml/Serialization/Compilation.cs

```diff
@@ -393,27 +393,12 @@ internal static bool GenerateSerializerFile(XmlMapping[] xmlMappings, Type[] typ
                 readerCodeGen.GenerateSerializerContract("XmlSerializerContract", xmlMappings, types, readerClass, readMethodNames, writerClass, writeMethodNames, serializers);
                 writer.Indent--;
                 writer.WriteLine("}");
-                string serializerName = XmlSerializer.GetXmlSerializerAssemblyName(types[0], null);
-                string location = Path.Combine(codePath, serializerName + ".cs");
-                try
-                {
-                    if (File.Exists(location))
-                    {
-                        File.Delete(location);
-                    }
 
-                    using (FileStream fs = File.Create(location))
-                    {
-                        string codecontent = compiler.Source.ToString();
-                        Byte[] info = new UTF8Encoding(true).GetBytes(codecontent);
-                        fs.Write(info, 0, info.Length);
-                        return true;
-                    }
-                }
-                catch (UnauthorizedAccessException)
-                {
-                    throw new UnauthorizedAccessException(SR.Format(SR.DirectoryAccessDenied, location));
-                }
+                string codecontent = compiler.Source.ToString();
```

Comment:
Good suggestion. Discussed with Shin, it is more easy to debug the code using StringWriter. We can consider making this change later.